src/main/groovy/org/shipkit/internal/gradle/versionupgrade/VersionUpgradeProducerPlugin.java

```diff
@@ -0,0 +1,125 @@
+ *
+ */
+public class VersionUpgradeProducerPlugin implements Plugin<Project> {
+
+    private VersionUpgradeProducerExtension versionUpgrade;
+
+    @Override
+    public void apply(final Project project) {
+        final ShipkitConfiguration conf = project.getPlugins().apply(ShipkitConfigurationPlugin.class).getConfiguration();
+
+        versionUpgrade = project.getExtensions().create("versionUpgradeProducer", VersionUpgradeProducerExtension.class);
+
+        final Task performAllUpdates = TaskMaker.task(project, "produceVersionUpgrade", new Action<Task>() {
+            @Override
+            public void execute(final Task task) {
+                task.setDescription("Performs dependency upgrade in all consumer repositories.");
+            }
+        });
+
+        DeferredConfiguration.deferredConfiguration(project, new Runnable() {
+            @Override
+            public void run() {
+                notNull(versionUpgrade.getConsumersRepositoriesNames(),
+                    "'versionUpgradeProducer.consumersRepositoriesName'");
+                for(String consumerRepositoryName : versionUpgrade.getConsumersRepositoriesNames()){
+                    Task cloneTask = createConsumerCloneTask(project, conf, consumerRepositoryName);
+                    Task performUpdate = createProduceUpgradeTask(project, consumerRepositoryName);
+                    performUpdate.dependsOn(cloneTask);
+                    performAllUpdates.dependsOn(performUpdate);
+                }
+            }
+        });
+    }
+
+    private Task createConsumerCloneTask(final Project project, final ShipkitConfiguration conf, final String consumerRepository){
+        return TaskMaker.task(project,
+            "cloneConsumerRepo" + cleanRepositoryName(consumerRepository),
+            CloneGitRepositoryTask.class,
+            new Action<CloneGitRepositoryTask>() {
+                @Override
+                public void execute(final CloneGitRepositoryTask task) {
+                    task.setDescription("Clones consumer repo " + consumerRepository + " into a temporary directory.");
+                    String gitHubUrl = conf.getGitHub().getUrl();
+                    task.setRepositoryUrl(gitHubUrl + "/" + consumerRepository);
+                    task.setTargetDir(getConsumerRepoTempDir(project, consumerRepository));
+                }
+        });
+    }
+
+    private Task createProduceUpgradeTask(final Project project, final String consumerRepository){
+        return TaskMaker.execTask(project, "produceVersionUpgrade" + cleanRepositoryName(consumerRepository), new Action<Exec>() {
+            @Override
+            public void execute(final Exec task) {
+                task.setDescription("Performs dependency upgrade in " + consumerRepository);
+                task.setWorkingDir(getConsumerRepoTempDir(project, consumerRepository));
+                task.commandLine("./gradlew", "performVersionUpgrade", getDependencyProperty(project));
```

Comment:
Yeah, version-upgrade-consumer. Do you think we can do better about error message, when plugin is not applied, than "Task :performVersionUpgrade not found"?